patches/net/minecraft/world/item/crafting/Ingredient.java.patch

```diff
@@ -78,11 +79,26 @@
 +     */
 +    public boolean synchronizeWithContents() {
 +        return true;
++    }
++
++    private boolean areAllStacksEmpty() {
++        Boolean empty = this.areAllStacksEmpty;
```

Comment:
I believe rather than having this local variable, we can just check `if (this.areAllStacksEmpty == null)` and then do `this.areAllStacksEmpty = allEmpty;` and `return this.areAllStacksEmpty`?

Reply:
I used this coding pattern to avoid reading the non-final field multiple times. May not make a difference to the JIT's ability to optimize, but doesn't hurt IMO.

Reply:
That pattern is generally only used for volatile variables, or in case of concurrent mutations.